src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int COEFFICIENT_FOR_INCREASE = 2;
+    private int size;
+    private Node<K, V>[] tableOfNodes;
+
+    public MyHashMap() {
+        tableOfNodes = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    private void increaseSize() {
+        size = 0;
+        Node<K, V>[] oldBuckets = tableOfNodes;
+        tableOfNodes = new Node[tableOfNodes.length * COEFFICIENT_FOR_INCREASE];
+        for (Node<K, V> node : oldBuckets) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> newNode = new Node<>(key, value);
+        int threshold = (int) (tableOfNodes.length * DEFAULT_LOAD_FACTOR);
+        if (threshold == size) {
+            increaseSize();
+        }
```

Comment:
You can calculate threshold in increaseSize method, maybe it will make some sense and put this checking in the beginning. But it's just a proposal, everything looks good